worker/app/operations/upload.py

```diff
@@ -30,9 +40,52 @@ def execute(self):
 
         try:
             with SFTPClient(hostname, port, username, private_key) as client:
-                for file in self.working_dir.iterdir():
+                for file in self.files:
                     if file.is_dir():
                         continue
                     client.upload_file(self.remote_working_dir, file)
+
+                    if self.delete:
+                        file.unlink()
         except SSHException as e:
             raise UploadError(code='paramiko.Error', message=str(e))
+
+    @staticmethod
+    def upload(remote_working_dir: str, files: list = None, directory: Path = None, delete: bool = False):
```

Comment:
Personally I am not a fan of having `files` and `directory` and ask user to provide one of them. What happens if I provide both, do they / should they all get uploaded? 

Maybe we should have `upload_files` and `upload_directory_content`, one could call the other? What do you think?

Reply:
👍